src/ximea_ros_cam.cpp

```diff
@@ -110,6 +114,23 @@ void XimeaROSCam::initNodeHandles() {
     ROS_INFO("... Node Handles Loaded. ");
 }
 
+void XimeaROSCam::initDiagnostics() {
+    if (this->enable_diagnostics) {
+        this->frequency_min = this->cam_framerate_set_ - this->frequency_tolerance;
```

Comment:
Currently I'm using the `this->cam_framerate_set_` parameter, but that may only be valid if `frame_rate_control: true`. How is the frequency regulated in this driver?

Reply:
It is regulated based on when the camera has an image in the buffer, so it is entirely camera based.

I plan on adding some actual polling duration for the camera images to decrease some cpu loading

Reply:
ok, I'll just add a parameter for the fps limits, and let the user set them as needed.